opentelemetry-exporter-geneva/geneva-uploader/src/ingestion_service/uploader.rs

```diff
@@ -0,0 +1,212 @@
+    #[allow(dead_code)]
+    fn create_upload_uri(&self, data_size: usize, event_name: &str, event_version: &str) -> String {
+        let now: DateTime<Utc> = Utc::now(); //TODO - this need to be calculated from the bond data
+        let end_time = now + ChronoDuration::minutes(5); //TODO - this need to be calculated from the bond data
+
+        // Format times in ISO 8601 format with fixed precision
+        // Using .NET compatible format (matches DateTime.ToString("O"))
+
+        let start_time = format!(
+            "{:04}-{:02}-{:02}T{:02}:{:02}:{:02}.{:07}Z",
+            now.year(),
+            now.month(),
+            now.day(),
+            now.hour(),
+            now.minute(),
+            now.second(),
+            now.nanosecond() / 100 // Convert nanoseconds to 7-digit precision
+        );
+
+        let end_time = format!(
+            "{:04}-{:02}-{:02}T{:02}:{:02}:{:02}.{:07}Z",
+            end_time.year(),
+            end_time.month(),
+            end_time.day(),
+            end_time.hour(),
+            end_time.minute(),
+            end_time.second(),
+            end_time.nanosecond() / 100 // Convert nanoseconds to 7-digit precision
+        );
+
+        // URL encode parameters
+        // TODO - Maintain this as url-encoded in config service to avoid conversion here
+        let encoded_monitoring_endpoint: String =
+            byte_serialize(self.monitoring_endpoint.as_bytes()).collect();
+
+        let encoded_source_identity: String =
+            byte_serialize(self.config.source_identity.as_bytes()).collect();
+
+        // Create a source unique ID - using a UUID to ensure uniqueness
+        let source_unique_id = Uuid::new_v4().to_string();
+
+        // Create the query string
+        format!(
+            "api/v1/ingestion/ingest?endpoint={}&moniker={}&namespace={}&event={}&version={}&sourceUniqueId={}&sourceIdentity={}&startTime={}&endTime={}&format={}&dataSize={}&minLevel={}&schemaIds={}",
+            encoded_monitoring_endpoint,
+            self.moniker,
+            self.config.namespace,
+            event_name,
+            event_version,
+            source_unique_id,
+            encoded_source_identity, // source identity (tenant/role/instance)
+            start_time,
+            end_time,
+            "centralbond/lz4hc", // Format encoding
+            data_size,
+            2, // Min level
+            self.config.schema_ids
+        )
+    }
+
+    /// Uploads data to the ingestion gateway
+    ///
+    /// # Arguments
+    /// * `data` - The encoded data to upload (already in the required format)
+    ///
+    /// # Returns
+    /// * `Result<IngestionResponse>` - The response containing the ticket ID or an error
+    #[allow(dead_code)]
+    pub(crate) async fn upload(
+        &self,
+        data: Vec<u8>,
+        event_name: &str,
+        event_version: &str,
+    ) -> Result<IngestionResponse> {
+        let data_size = data.len();
```

Comment:
Should we check for `data.len() > 0` here to avoid uploading empty data?

Reply:
This would be handled in the public Uploader API - which will accept OTLP payload, and convert internally into the bond encoding before invoking this crate-provide API.